File: command/typescript/buy-stock.ts
import {IOrderCommand} from "./order-command.interface";
import {StockManager} from "./stock-manager";

// Nesnelere dönüştürdüğümüz her bir isteğe denk gelmektedir.
// Command (IOrderCommand) arayüzünü uygular.
// UML diyagramındaki ConcreteCommand sınıfına denk gelmektedir.
export class BuyStock implements IOrderCommand{
  constructor(private readonly stockManager: StockManager) {}

  public execute(): void {
    this.stockManager.buy();
  }
}

File: command/go/buy-stock.go
package main

// Nesnelere dönüştürdüğümüz her bir isteğe denk gelmektedir.
// UML diyagramındaki ConcreteCommand sınıfına denk gelmektedir.
type buyStock struct {
	sm *StockManager
}

func NewBuyStock(sm *StockManager) *buyStock {
	return &buyStock{
		sm: sm,
	}
}

func (s buyStock) Execute() {
	s.sm.Buy()
}
